src/NuGet.Services.KeyVault/CachingSecretReader.cs

```diff
@@ -44,8 +57,16 @@ public async Task<ISecret> GetSecretObjectAsync(string secretName)
             {
                 return result.Secret;
             }
+
+            var start = DateTimeOffset.UtcNow;
             // The cache does not contain a fresh copy of the secret. Fetch and cache the secret.
             var updatedValue = new CachedSecret(await _internalReader.GetSecretObjectAsync(secretName));
+
+            logger?.LogInformation("Refresh secret {SecretName}{ExpirationTime} (Latency = {ElapsedMilliseconds}ms)",
+                updatedValue.Secret.Name,
+                updatedValue.Secret.Expiration == null ? "" : " which expires at " + ((DateTimeOffset) updatedValue.Secret.Expiration).UtcDateTime,
+                (DateTimeOffset.UtcNow - start).TotalMilliseconds.ToString("F2"));
```

Comment:
We can search and analyze the log values using App Insights. One may want to do some number crunching on the `ExpirationTime` parameter, but this is difficult to do if the values are prefixed with `which expired at `. Maybe something like this instead:

```suggestion
            logger?.LogInformation("Refreshed secret {SecretName} in {ElapsedMilliseconds}ms, expires at {ExpirationTime}",
                updatedValue.Secret.Name,
                (DateTimeOffset.UtcNow - start).TotalMilliseconds.ToString("F2"),
                updatedValue.Secret.Expiration == null
                    ? DateTimeOffset.MaxValue
                    : ((DateTimeOffset) updatedValue.Secret.Expiration).UtcDateTime);
```

Reply:
Yeah, I have updated the log info, and I may prefer to log "null" rather than DateTimeOffset.MaxValue for null expiration time.